tableauserverclient/server/endpoint/endpoint.py

```diff
@@ -64,9 +69,17 @@ def _make_request(
 
         logger.debug("request {}, url: {}".format(method.__name__, url))
         if content:
-            logger.debug("request content: {}".format(content[:1000]))
+            logger.debug("request content: %r", content[:1000])
 
         server_response = method(url, **parameters)
+
+        # Response.content is a property. Calling it will load the entire response into memory. Checking if the
+        # content-type is an octet-stream accomplishes the same goal without eagerly loading content.
+        stream = parameters.get("stream", False)
+        stream = stream or (server_response.headers.get("Content-Type") == "application/octet-stream")
+        if stream:
+            return server_response
```

Comment:
don't we still want to check the status of the response? the _check_status method as is does refer to response.content, but that could be changed.

Reply:
Reordered things so the response status code is properly checked before returning the stream.